### ISSUE TITLE
configure script is not executable

### ISSUE THREAD

**Issue by fingolfin** [simpcomp-team/simpcomp#8]
The configure script in your release tarball is missing the executable bit. As a result, the package cannot be built, unless one sets that flag. The `BuildPackages.sh` will not pick the package up for building either. So people who build GAP from source using this tool (and I imagine that's quite a lot of them) will never see the binaries in simpcomp being compiled.

**Comment by fingolfin** [comment]
Actually, this affects more files: depcomp install-sh missing

**Comment by simpcomp-team** [comment]
Made "configure" executable and pushed changes to github. BuildPackages.sh now compiles source code and puts it into simpcomp/bin if

    simpcomp*)
      echo_run ./configure
      echo_run make
      echo_run make install
      echo_run make clean
    ;;

is added to "BuildPackages.sh".

Will try to release new version (2.1.9) as soon as possible.

**Comment by simpcomp-team** [comment]
Wouldn't it be best to be compatible with `run_configure_and_make` in `BuildPackages.sh`?

In this case, am I correct to assume that all I need is to add a file `autogen.sh`

    #!/bin/sh -ex
    #
    # simpcomp package for GAP
    #
    # This file is part of the build system of a GAP kernel extension.
    # Requires GNU autoconf, GNU automake and GNU libtool.
    #
    autoreconf -vif `dirname "$0"`

and then

    simpcomp*)
      echo_run ./configure
      echo_run make
      echo_run make install
      echo_run make clean
    ;;

can be removed from `BuildPackages.sh`?

**Comment by fingolfin** [comment]
If you mean that you'd add that file and at the same time, remove all generated files (like `configure`), then: yes :-).

**Comment by simpcomp-team** [comment]
One more stupid question: does `BuildPackages.sh` also run `make install`? Our executable `bistellar` needs to be copied into the `bin` directory.

**Comment by fingolfin** [comment]
Nope, `make install` is not run -- it is expected that `make` takes care of copying files to `bin`. (After all, `make install` should install into the `$prefix` dir, which usually is *not* the `$srcdir`). In the future, we plan to implement `make install` for GAP itself, at which point it'll also become useful for packages to implement it in a certain way -- but it is not yet quite clear how that will work exactly.

**Comment by fingolfin** [comment]
BTW, I think this was closed by accident when the gh-pages stuff was merged :-(

**Comment by simpcomp-team** [comment]
Ok, just double checking: So you want me to manually add something like

    all-local:
      mv bistellar bin/bistellar

to `Makefile.am`?

**Comment by fingolfin** [comment]
That would be rather brittle, as it'll break if `bistellar` has not yet been compiled, or if there is no` bin` dir. Perhaps this:
```
all-local: bistellar
	mkdir -p bin
	cp bistellar bin/bistellar
```

**Comment by fingolfin** [comment]
I put this into PR #13. Once that is merged, I think this issue is fully resolved, thank you. I'll work on a PR for GAP that adjust `BuildPackage.sh` (I'll try to make it so that it works with both this new simpcomp and also older versions).

**Comment by fingolfin** [comment]
Fixed now, thanks
